plugins/animate/squeezimize.hpp

```diff
@@ -276,15 +328,49 @@ class squeezimize_transformer : public wf::scene::view_2d_transformer_t
                 (this->minimize_target.y + this->minimize_target.height) - bbox.y),
                 (bbox.y + bbox.height) - this->minimize_target.y);
 
+        bool horiz;
+        // auto src_box = view->get_bounding_box();
+        auto output = view->get_output();
+        auto geom   = output->get_relative_geometry();
+
+        // check which edge the target is closest to
+        double x  = minimize_target.x + minimize_target.width / 2.0;
+        double y  = minimize_target.y + minimize_target.height / 2.0;
+        double y2 = y * geom.width / geom.height;
+        if (x < y2)
+        {
+            // bottom left part of the screen
+            if (x < geom.width - y2)
+            {
+                // left edge
+                horiz = true;
+                this->upward = true;
+            } else
+            {
+                // bottom edge
+                horiz = false;
+                this->upward = false;
+            }
+        } else
+        {
+            // top right part of the screen
+            if (x > geom.width - y2)
+            {
+                // right edge
+                horiz = true;
+                this->upward = false;
+            } else
+            {
+                // top edge
+                horiz = false;
+                this->upward = true;
+            }
+        }
+
         wf::gles::run_in_context_if_gles([&]
         {
-            program.compile(squeeze_vert_source, squeeze_frag_source);
+            program.compile(squeeze_vert_source, horiz ? squeeze_frag_source_horiz : squeeze_frag_source);
```

Comment:
We could have two programs compiled upfront and ready to use, but this is probably fine too, since the shaders are cached.